src/utils/math.hpp

```diff
@@ -64,4 +67,44 @@ constexpr std::optional<uint64_t> RoundUint64ToMultiple(uint64_t val, uint64_t m
   return (numerator / multiple) * multiple;
 }
 
+/*!
+ * @return The default precision used in the project. Two doubles are considered different if the distance between them
+ * is strictly greater than Epsilon.
+ */
+constexpr double GetEpsilon() noexcept { return 0.0000001; }
+
+/*!
+ *
+ * @param a
+ * @param b
+ * @param epsilon The precision to use when comparing the two double. If dist(a,b)<=epsilon then a==b.
+ * @return an integer indicating whether a and b are equal.
+ *          +0: dist(a,b) <= epsilon
+ *          +1: a > b+epsilon
+ *          -1: a < b-epsilon
+ */
+constexpr int16_t CompareDouble(double a, double b, double epsilon = GetEpsilon()) noexcept {
```

Comment:
Also, could you plz expand on why we need this? I get there is a precision error somewhere, but what exactly is it and how does it happen?

Reply:
> Also, could you plz expand on why we need this? I get there is a precision error because of the `doubles`, but what exactly is it and how does it happen? Can you give me an example?

Interesting question.

1. Overall one should not compare double as follow `myDouble == myOtherDouble`  because of the precision behind double. The following UT can fail for that reason:
```
TEST(UtilsMath, example) {
  // x = (value in arithmetic, value on computer) - https://www.binaryconvert.com/convert_double.html
  double a = 42.424242424242424242;           // a = (42.424242424242422, 42.4242424242424220892644370906)
  int b = 1 / memgraph::utils::GetEpsilon();  // b = 10000000
  double c = a / b;                           // c = (0,000004242, 0.00000424200000000000021249972267934 )
  double d = 0.0;                             // d = (0.0, 0.0)
  for (auto idx = 0; idx < b; ++idx) {
    d += c;
  }
  // d = (42.424242424242422, 42.4242424201029919572647486348)
  EXPECT_TRUE(memgraph::utils::CompareDouble(a, d) == 0);
  EXPECT_TRUE(a == d); // can fail
}
```
2. I do not need to check whether `a == b` in my case but `a <= b`, which can lead to the same problem:
-   `a= 1.0; b = 100.0;` so  `a < b` is `true`. Expected.
-   `a = 1.000000000000001; b = 1.0000000000000011;` so `a < b` is `true`. Is it really expected, is the difference due to the double rounding? Maybe we want to consider them equal in that case.
-   `a = 1.0000000000000011; b = 1.000000000000001;` so `a < b is false`. Same as above.

3. I realize we have problematic comparison in the code-base, for instance:
https://github.com/memgraph/memgraph/blob/7a2bbd4bb37920322a0097d964c2751027602e55/src/query/typed_value.cpp#L742

But this goes beyond the scope of this PR, I have created a task on Airtable to tackle it properly and independently.

Reply:
After discussion with the whole team: we will not keep it for now until it is proven to be a problem. The responsibility of comparing double correctly is on the side of the user and Memgraph only mimic the c++ comparison.